ansible_base/lib/abstract_models/common.py

```diff
@@ -62,10 +62,13 @@ def save(self, *args, **kwargs):
         '''
         update_fields = list(kwargs.get('update_fields', []))
 
-        if 'modified_by' not in update_fields:
+        if self.pk and 'modified_by' not in update_fields:
             self.modified_by = current_user_or_system_user()
             update_fields.append('modified_by')
 
+        if kwargs.get('update_fields') is not None:
+            kwargs['update_fields'] = update_fields
```

Comment:
There was a time when the prior definition of `update_fields =` did not do `list()`, in which case, this step is not needed. It's very subtle, but works. It's not surprising that this regressed.